comm/src/main/scala/coop/rchain/comm/rp/Connect.scala

```diff
@@ -28,6 +28,12 @@ object Connect {
 
   object ConnectionsCell {
     def apply[F[_]](implicit ev: ConnectionsCell[F]): ConnectionsCell[F] = ev
+
+    def random[F[_]: Monad: ConnectionsCell: RPConfAsk]: F[Connections] =
```

Comment:
@KentShikama could you have a look on my comment above? Just to double check

Reply:
Requests for missing blocks are sent directly to the peer that requested it right?

Reply:
@KentShikama I guess yes. But @rabbitonweb 's concern is that the node could request the missing block from (a small number of random) peers that don't have the block yet. So it would never receive the missing block. Is that right?

Reply:
The call every X seconds to fetchDependencies ( https://github.com/rchain/rchain/blob/9ae5825aa8b0469372976873b7f229d73060d5fd/casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala#L499 ) should take care of it eventually.